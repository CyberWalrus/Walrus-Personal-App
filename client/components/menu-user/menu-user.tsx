import withUserMenuState from "@client/hocs/with-user-menu-state/with-user-menu-state";
import RoutePath from "@client/routes";
import { User } from "@client/type/data";
import { FormType } from "@config/constants";
import * as React from "react";
import { ComponentClass, FunctionComponent, ReactElement } from "react";
import { Link } from "react-router-dom";
import { compose } from "recompose";
import FormCustom from "../form-custom/form-custom";

export interface PropsHoc {
  isOpen: boolean;
  isAuthorization: boolean;
  user: User;
  onLogout: () => void;
  onClickMenu: () => void;
}
type Props = PropsHoc;
const MenuUser: FunctionComponent<Props> = ({
  isOpen,
  isAuthorization,
  user,
  onClickMenu,
  onLogout,
}: Props): ReactElement => {
  return (
    <div className={`menu-user-box`}>
      <button onClick={onClickMenu} className={`menu-user-box__btn`}>
        {isAuthorization ? (
          <React.Fragment>{user ? user.login : ``}</React.Fragment>
        ) : (
          `SignIn/SignUp`
        )}
      </button>
      <div className={isOpen ? `menu-user` : `menu-user_close`}>
        {isAuthorization ? (
          <div className={`user`}>
            <Link to={RoutePath.OPTION} className={`user__link`}>
              Option
            </Link>
            <button onClick={onLogout} className={`menu-user-box__btn`}>
              Logout
            </button>
          </div>
        ) : (
          <div className={`user`}>
            <FormCustom formType={FormType.SIGN_IN} titel={`Sign In`} />
            <Link to={RoutePath.SIGNUP} className={`user__link`}>
              Sign Up
            </Link>
          </div>
        )}
      </div>
    </div>
  );
};

export { MenuUser };

const wrapper = compose(withUserMenuState);

export default wrapper(MenuUser) as ComponentClass;
