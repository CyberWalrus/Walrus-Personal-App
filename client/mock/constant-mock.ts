import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
