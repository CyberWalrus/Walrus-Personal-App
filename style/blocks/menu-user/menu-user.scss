@import "./__arrow/menu-user__arrow.scss";
@import "./__item/menu-user__item.scss";
@import "./__link/menu-user__link.scss";
@import "./__list/menu-user__list.scss";
@import "./__nav/menu-user__nav.scss";

.menu-user {
  margin: 0rem;
  padding: 0rem;
  position: fixed;
  z-index: 999;
  right: 0rem;
  bottom: 4rem;
  max-inline-size: 30rem;
  display: inline-block;
  visibility: hidden;
  opacity: 0;
  border-radius: 3rem;
  @include transition(all 0.15s);
  &_open {
    visibility: visible;
    opacity: 1;
  }
}
