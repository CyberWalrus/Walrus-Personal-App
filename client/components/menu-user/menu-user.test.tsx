import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { user } from "../../mock/constant-mock";
import { MenuUser } from "./menu-user";

Enzyme.configure({ adapter: new Adapter() });
it(`MenuUser correctly render without data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(
      <MenuUser
        isOpen={true}
        isAuthorization={false}
        user={undefined}
        onClickMenu={jestFunc}
        onLogout={jestFunc}
        onButtonCLick={jestFunc}
      />,
    ),
  );
  expect(tree).toMatchSnapshot();
});
it(`MenuUser correctly render with data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(
      <MenuUser
        isOpen={true}
        isAuthorization={false}
        user={user}
        onClickMenu={jestFunc}
        onLogout={jestFunc}
        onButtonCLick={jestFunc}
      />,
    ),
  );
  expect(tree).toMatchSnapshot();
});
