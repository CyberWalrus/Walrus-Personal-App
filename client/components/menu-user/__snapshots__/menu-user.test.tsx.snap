// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuUser correctly render with data 1`] = `
<div
  className="menu-box"
>
  <button
    className="menu-box__btn menu-box__btn_active"
    onClick={[MockFunction]}
  >
    t
  </button>
  <div
    className="menu-user menu-user_open"
  >
    <div
      className="menu-user__box"
    >
      <nav
        className="menu-user__nav"
      >
        <ul
          className="menu-user__ul"
        >
          <li
            className="menu-user__item"
          >
            <ContainerButton
              onButtonCLick={[MockFunction]}
            />
          </li>
          <li
            className="menu-user__item"
          >
            <Connect(WithFormState)
              formType="SIGN_IN"
              isTitel={false}
              titel="Sign In"
            />
          </li>
          <li
            className="menu-user__item"
          >
            <Link
              className="menu-user__link"
              to="/signup"
            >
              <div
                className="menu-user__icon"
              >
                N
              </div>
              <div
                className="menu-user__text"
              >
                SignUp
              </div>
            </Link>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="menu-user__arrow"
    />
  </div>
</div>
`;

exports[`MenuUser correctly render without data 1`] = `
<div
  className="menu-box"
>
  <button
    className="menu-box__btn menu-box__btn_active"
    onClick={[MockFunction]}
  >
    t
  </button>
  <div
    className="menu-user menu-user_open"
  >
    <div
      className="menu-user__box"
    >
      <nav
        className="menu-user__nav"
      >
        <ul
          className="menu-user__ul"
        >
          <li
            className="menu-user__item"
          >
            <ContainerButton
              onButtonCLick={[MockFunction]}
            />
          </li>
          <li
            className="menu-user__item"
          >
            <Connect(WithFormState)
              formType="SIGN_IN"
              isTitel={false}
              titel="Sign In"
            />
          </li>
          <li
            className="menu-user__item"
          >
            <Link
              className="menu-user__link"
              to="/signup"
            >
              <div
                className="menu-user__icon"
              >
                N
              </div>
              <div
                className="menu-user__text"
              >
                SignUp
              </div>
            </Link>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="menu-user__arrow"
    />
  </div>
</div>
`;
